### PR TITLE
#3253 [Manager] Using a better way to check configurator is available

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3253)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
refactor
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/utils/configHelpers.js [hunk 1]
```diff
@@ -43,8 +43,29 @@ const validateUrl = async url => {
 
   // Ensure the API URL can be reach
```
Comment (konekoya) [inline ohara-manager/utils/configHelpers.js]: Oops, my own typo...
```suggestion
  // Ensure the API URL can be reached
```

File: ohara-manager/utils/configHelpers.js [hunk 2]
```diff
@@ -155,8 +158,8 @@ const getConfig = () => {
       .version(false) // Disable version info, see https://github.com/oharastream/ohara/issues/1892
       .check(argv => {
         const { configurator, port } = argv;
-        validateUrl(configurator);
         validatePort(port);
+        validateUrl(configurator);
```
Comment (konekoya) [inline ohara-manager/utils/configHelpers.js]: Does the call order matter here? Just wondering :)

File: ohara-manager/utils/configHelpers.js (hunk 1 above)
Comment (saivirtue) [reply ohara-manager/utils/configHelpers.js]: had been edited :)

File: ohara-manager/utils/configHelpers.js (hunk 2 above)
Comment (saivirtue) [reply ohara-manager/utils/configHelpers.js]: Hmm...I think they should have same result. 😆

File: ohara-manager/utils/configHelpers.js [hunk 3]
```diff
@@ -41,10 +41,31 @@ const validateUrl = async url => {
...
+    const { hostname, port } = new URL(url);
+
+    return new Promise(resolve => {
+      const socket = new net.Socket();
+      const onError = err => {
+        console.log(err.message);
+        socket.destroy();
+        resolve(false);
+      };
+      socket.setTimeout(10000);
```
Comment (chia7712) [inline ohara-manager/utils/configHelpers.js]: 這有包含retry的邏輯嗎？

Comment (saivirtue) [reply ohara-manager/utils/configHelpers.js]: 這一段沒有，純粹只是設定timeout；retry邏輯在`while()`部分

**Comment by chia7712** [comment]
現在這個方式會變成只有確保tcp/ip是通的，而之前是有驗證到http server也已經啟動完成。不過目前兩者在Configurator的啟動理論上是「相當接近」，所以不會有太大的問題

**Pull request merged**